trl/trainer/kto_trainer.py

```diff
@@ -1027,6 +1008,9 @@ def compute_loss(
 
         loss, metrics = self.get_batch_loss_metrics(model, inputs, train_eval="train")
 
+        # Make sure to move the loss to the device the original accumulating loss is at back in the `Trainer` class:
+        loss = loss.to(self.args.device)
```

Comment:
@kashif if your question was regarding this line:  
Without this I get an exception when running on multiple GPUs: 
`ValueError: Calculated loss must be on the original device: cuda:0 but device in use is cuda:1`.  
I copied this line from dpo_trainer to fix this, see [dpo_trainer.py#L1084](https://github.com/huggingface/trl/blob/main/trl/trainer/dpo_trainer.py#L1084)